ts/private/ts_project_worker.js

```diff
@@ -243,10 +244,22 @@ function createProgram(args, initialInputs) {
             return { close: () => {} };
         }
         debuglog(`watchDirectory ${directoryPath}`);
-        directoryWatchers.set(directoryPath, (input) => callback(path.join(execRoot, input)));
+
+        const cb = (input) => callback(path.join(execRoot, input))
+
+        const callbacks = directoryWatchers.get(directoryPath) || new Set();
+        callbacks.add(cb)
+        directoryWatchers.set(directoryPath, callbacks);
+
         return {
             close: () => {
-                directoryWatchers.delete(directoryPath);
+                const callbacks = directoryWatchers.get(directoryPath);
+                callbacks.delete(cb);
+                if (callbacks.size == 0) {
+                    directoryWatchers.delete(directoryPath)
+                } else {
+                    directoryWatchers.set(directoryPath, callbacks)
```

Comment:
I don't think we need to set it again?

Reply:
You are right. but wouldn't it help you reason about the whole directory watcher logic?

Reply:
I don't think so, it's just confusing imo 🤷

Reply:
I'll remove then